evalml/pipelines/components/transformers/feature_selection/feature_selector.py

```diff
@@ -10,61 +14,35 @@ def get_names(self):
         """Get names of selected features.
 
         Returns:
-            list of the names of features selected
+            List of the names of features selected
         """
         selected_masks = self._component_obj.get_support()
         return [feature_name for (selected, feature_name) in zip(selected_masks, self.input_feature_names) if selected]
 
     def transform(self, X, y=None):
-        """Transforms data X by selecting features
+        """Transforms input data by selecting features.
 
         Arguments:
-            X (pd.DataFrame): Data to transform
-            y (pd.Series, optional): Target data
+            X (ww.DataTable, pd.DataFrame): Data to transform
+            y (ww.DataColumn, pd.Series, optional): Target data. Ignored.
 
         Returns:
-            pd.DataFrame: Transformed X
+            ww.DataTable: Transformed X
         """
-        if isinstance(X, pd.DataFrame):
-            self.input_feature_names = list(X.columns.values)
-        else:
-            self.input_feature_names = range(X.shape[1])
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        self.input_feature_names = list(X.columns.values)
 
         try:
             X_t = self._component_obj.transform(X)
         except AttributeError:
-            raise RuntimeError("Transformer requires a transform method or a component_obj that implements transform")
-        if not isinstance(X_t, pd.DataFrame) and isinstance(X, pd.DataFrame):
-            X_dtypes = X.dtypes.to_dict()
-            selected_col_names = self.get_names()
-            col_types = {key: X_dtypes[key] for key in selected_col_names}
-            return pd.DataFrame(X_t, columns=selected_col_names, index=X.index).astype(col_types)
-        else:
-            return pd.DataFrame(X_t)
+            raise RuntimeError("Feature selector requires a transform method or a component_obj that implements transform")
```

Comment:
Yup, you're right! Updated this to return `evalml.exceptions.exceptions.MethodPropertyNotFoundError`, since that's more consistent with our other transformers.